### PR TITLE
relationships and new classes implementation

### PR DESCRIPTION
**Pull request by alexansaa**
In this milestone, the following features have been implemented:
- Developed Classroom, Book, and Rental classes
- Implemented relationships between Classroom and Student
- Implemented relationships between Rental, Person, and Book
- Updated classes for integration correctness

### CODE REVIEW

File: src/classroom.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Classrom
+  attr_accessor :label, :students
+
```
Comment (OyePriscilla) [inline src/classroom.rb]: - [ ] Kindly ensure that the `students` symbol can only retrieve the value of instance of the variable. Getters method should be implemented which can only provide read access to the attributes of an object. ℹ️
```suggestion
  attr_reader :students
```

File: src/students.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 require_relative 'person'
 
 class Student < Person
+  attr_accessor :classroom
+
```
Comment (OyePriscilla) [inline src/students.rb]: - [ ] Kindly ensure that the `classroom` symbol can only retrieve the value of instance of the variable. Getters method should be implemented which can only provide read access to the attributes of an object. ℹ️
```suggestion
  attr_reader :classroom
```

File: src/person.rb [hunk 3]
```diff
@@ -2,14 +2,15 @@
...
+  attr_accessor :name, :age, :rentals
 
   def initialize(id, name = 'Unknown', age = 18, parent_permission: true)
     super()
     @id = id
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (OyePriscilla) [inline src/person.rb:15]: - [ ] Please kindly implement a method to add rentals as stated in the requirement, kindly check this [link for better understanding](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship). 🙏 

g126

File: src/book.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (OyePriscilla) [inline src/book.rb]: - [ ] Please kindly implement a method to add rentals as stated in the requirement, kindly check this [link for better understanding](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship). 🙏 

g126

File: src/rental.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+  end
```
Comment (OyePriscilla) [inline src/rental.rb]: - [ ] Kindly make implementation for the rental that is adding itself to the @book and @person ** in order to achieve the 'belongs-to' association**

g126

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @alexansaa 👏 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Implemented a program using the command line. ✔️
Set up associations between class and object. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/classroom.rb (hunk 1 above)
Comment (alexansaa) [reply src/classroom.rb]: The suggestion is not clear enough. Do you mean to remove access to the "label" attribute as part of the suggestion or to make the "students" array just readable, and let "label" as it is now? Also, I see that you are talking about the "students" object when the marked code is in the "classroom" object code.

File: src/students.rb (hunk 2 above)
Comment (alexansaa) [reply src/students.rb]: Again, no clarity in the suggestion. You are talking about the classroom object when marking code from students object code

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @alexansaa 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

image


Highlights
- [x] Good project organization ✔️
- [x] Nice use of Ruby features ✔️ 
- [x] Beautiful class calls work 
- [x] Kudus on meeting the requirements   👏 


Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**